contrib/update-submodules.sh

```diff
@@ -8,4 +8,4 @@ WORKDIR=$(readlink -f "${WORKDIR}")
 "$WORKDIR/sparse-checkout/setup-sparse-checkout.sh"
 git submodule init
 git submodule sync
-git submodule update --depth=1
+git submodule update --depth=1 --jobs 16
```

Comment:
This doesn't work as well as it could

To address it even better, we could reuse the logic from https://github.com/ClickHouse/checkout/blob/main/action.yaml#L51-L55

Reply:
Amazing, this trick brought down runtime further from 258 sec to 91 sec.